### PR TITLE
Add configurable list of classmethod / staticmethod decorators

### LINKED ISSUE #38: Config to allow decorators that function like @classmethod
In SQLAlchemy, there is a `declared_attr` decorator that creates class methods:
http://docs.sqlalchemy.org/en/latest/orm/extensions/declarative/api.html#sqlalchemy.ext.declarative.declared_attr

As a result, my codebase is sprinkled with:
```
class SomeTable(db.Model):
    @declared_attr
    def __table_args__(cls):  # noqa: N805
        ...
```

It would be nice if we could have a config option that lets us declare if a decorator results in a `staticmethod` or `classmethod` as opposed to the implicit default of it being an instance method. I was thinking something like using comma separated list in the setup.cfg/tox.ini file, e.g:

```
[flake8]
classmethod-decorators=declared_attr
staticmethod-decorators=staticdeco1,staticdeco2
```

### PR DESCRIPTION
**Pull request by saifelse**
- Add classmethod-decorator and staticmethod-decorator config parameters
- Add support for parsing CLI-style flake8-config parameters in testing
  framework.
- Remove support for multiple `codes` in a testcase, since it was never used.
- Add tests for late-decorated methods

Fixes #38

### CODE REVIEW

File: pep8ext_naming.py [hunk 1]
```diff
@@ -98,11 +120,35 @@ def add_options(cls, parser):
                      help='List of names the pep8-naming plugin should '
                           'ignore. (Defaults to %default)')
 
+        classmethod_decorators = ','.join(_default_classmethod_decorators)
```
Comment (sigmavirus24) [inline pep8ext_naming.py]: This should be sorted so the output is reproducible.

File: pep8ext_naming.py [hunk 2]
```diff
@@ -98,11 +120,35 @@ def add_options(cls, parser):
...
+        register_opt(parser, '--classmethod-decorators',
+                     default=classmethod_decorators,
+                     action='store',
+                     type='string',
+                     parse_from_config=True,
+                     comma_separated_list=True,
+                     help='List of method decorators pep8-naming plugin '
+                          'should consider classmethod (Defaults to %default)')
+
+        staticmethod_decorators = ','.join(_default_staticmethod_decorators)
```
Comment (sigmavirus24) [inline pep8ext_naming.py]: This should also be sorted.

File: run_tests.py [hunk 3]
```diff
@@ -1,14 +1,21 @@
-import sys
 import os
-import pep8ext_naming
 import re
+import sys
+
+import flake8
+from flake8.options.manager import OptionManager
```
Comment (sigmavirus24) [inline run_tests.py]: You should not be importing this. It's a private implementation detail of Flake8 that may change location.

**Comment by saifelse** [comment]
> This is a fantastic start @saifelse

@sigmavirus24: Thanks! Note that the latest update is broken, but works after applying [a bugfix for flake8_polyfill](https://gitlab.com/pycqa/flake8-polyfill/merge_requests/4)

**Comment by saifelse** [comment]
@sigmavirus24: just checking in to see if you've had a chance to look at this PR / [bugfix for flake8_polyfill](https://gitlab.com/pycqa/flake8-polyfill/merge_requests/4). Cheers.

**Comment by saifelse** [comment]
@sigmavirus24 : checking in again to see if you've had a chance to look this over or the flake8_polyfill bugfix. Thanks!

**Comment by sigmavirus24** [comment]
Hey @saifelse I'm truly sorry it took me so long to work my way back around to this project/PR. The changes here look excellent. Can you rebase them? If not, just let me know and I'll tackle that myself.

**Comment by saifelse** [comment]
Hey @sigmavirus24, no worries! I rebased and fixed the merge conflicts.

Note that this is dependent on flake8-polyfill being republished for which I've opened a merge request: https://gitlab.com/pycqa/flake8-polyfill/merge_requests/6

**Comment by sigmavirus24** [comment]
Closing and then re-opening to test against 1.0.2 of flake8-polyfill

**Pull request merged**